### PR TITLE
Enable example testing

### PR DESCRIPTION
**Pull request by XiaokaiUber**
For #2331

#### Change List
- added new test examples
- fixed the golden images of examples that did not pass the test

### CODE REVIEW

**Comment by Pessimistress** [comment]
What's wrong with the react-app image

**Comment by XiaokaiUber** [comment]
> What's wrong with the react-app image

@Pessimistress 
It is a dropdown list which allows you to display different layers.

File: test/node-examples.js [hunk 1]
```diff
@@ -137,9 +139,17 @@ async function runTestExample(folder) {
   exampleDir = EXAMPLES_GET_STARTED_DIR;
   await runTestExample('pure-js');
   await runTestExample('pure-js-without-map');
-  // await runTestExample('react-webpack');
+  // await runTestExample('react-browserify');
+  await runTestExample('react-webpack-2');
   await runTestExample('react-without-map');
 
+  exampleDir = EXAMPLES_JSON_BROWSER_DIR;
+  await runTestExample('pure-js-app');
```
Comment (Pessimistress) [inline test/node-examples.js]: We are starting to run into name collisions (`pure-js` vs `pure-js-app`). The naming convention of golden images needs to be updated to include directory names.

**Comment by Pessimistress** [comment]
In the react-app golden image, the base map is not fully loaded. This is not going to work on other people's machines or networks.

File: test/node-examples.js [hunk 2]
```diff
@@ -131,24 +134,26 @@ async function runTestExample(folder) {
...
-  await runTestExample('brushing');
-  await runTestExample('geojson');
-  await runTestExample('icon');
-  await runTestExample('line');
-  await runTestExample('plot');
-  await runTestExample('scatterplot');
-  await runTestExample('screen-grid');
-  await runTestExample('tagmap');
+  exampleDir = EXAMPLES_DIR;
+  await runTestExample('experimental', 'bezier');
```
Comment (Pessimistress) [inline test/node-examples.js]: Why not just accept one argument e.g. `runTestExample('experimental/bezier')`?

**Comment by Pessimistress** [comment]
The moved json examples need to be named something meaningful e.g. `json-pure-js`, `json-react`

**Pull request merged**